Propagate quarkusRegistryClient property

jsmrcka opened the pull request:
The property tells Quarkus Maven Plugin whether it should use the online registry to resolve extension catalog. Default is `true` for Quarkus. Thanks to the propagation, it can now be overriden when running tests by using `-DquarkusRegistryClient=false`.

Fixes
https://github.com/quarkus-qe/quarkus-extensions-combinations/issues/115.

Additional fixes:
- Decrease combinations size in native GH test
  - To avoid OutOfMemoryError for increased number of extensions (174 at the
moment).
- Exclude camel-quarkus-mybatis and logging-json
  - `camel-quarkus-mybatis` needs a datasource.
  - There is a name collision between `io.quarkus:quarkus-logging-json` and `io.quarkiverse.loggingjson:quarkus-logging-json`.

jsmrcka committed:
Propagate quarkusRegistryClient property

The property tells Quarkus Maven Plugin whether it should use the online
registry to resolve extension catalog. Default is `true` for Quarkus.
Thanks to the propagation, it can now be overriden when running tests by
using `-DquarkusRegistryClient=false`.

Fixes
https://github.com/quarkus-qe/quarkus-extensions-combinations/issues/115.

jedla97 reviewed (approved):
Thanks for resolving it! LGTM!

jedla97 commented:
Not look on all tests but some of them failing because of `logging-json` which have two version 
```
- io.quarkus:quarkus-logging-json
- io.quarkiverse.loggingjson:quarkus-logging-json
```

jsmrcka committed:
Decrease combinations size in native GH test

To avoid OutOfMemoryError for increased number of extensions (174 at the
moment).

jsmrcka committed:
Exclude camel-quarkus-mybatis and logging-json

See the code for reasons.

rsvoboda commented:
@jsmrcka How is this gonna work when new branch is introduced? 

When we for example introduce 3.7 branch, will it fetch list of extensions for 3.7 Quarkus stream and main will be using Quarkus latest?

jsmrcka commented:
> @jsmrcka How is this gonna work when new branch is introduced?
> 
> When we for example introduce 3.7 branch, will it fetch list of extensions for 3.7 Quarkus stream and main will be using Quarkus latest?

@rsvoboda Extensions are being listed using this POM: https://github.com/quarkus-qe/quarkus-extensions-combinations/blob/main/src/main/resources/list-extensions.pom.xml, which imports a BOM based on root POM of the TS (or on mvn arguments, if provided).
Different branches usually have their own Quarkus version, so for each one, the list of extensions should correspond to the Quarkus stream. 

This PR does not change that.

rsvoboda commented:
So the command is operating on top of generated list-extensions.pom.xml, so the right stream gets picked from the bom defined there.

I still see some challenges:
Upstream:
- once the upstream removes stream entries from registry, tests won't work 
  - available streams: https://registry.quarkus.io/client/platforms
  - tbh, risk is pretty low as LTS streams should be available for a long time

Downstream:
 - planning to use dedicated registry or using `-DquarkusRegistryClient=false` ?

jsmrcka commented:
> Downstream:
> - planning to use dedicated registry or using -DquarkusRegistryClient=false ?

Yes, the ` -DquarkusRegistryClient=false`. See jenkins-jobs/-/merge_requests/747.

rsvoboda reviewed (approved):
CI is green now

rsvoboda commented:
Triggered https://github.com/quarkus-qe/quarkus-extensions-combinations/actions/runs/6967176672 to see the longer run
